devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,346 @@
+}
+
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping(endpoint = null) {
+	log.info "ping called"
+	refresh(endpoint)
+}
+
+def refresh(endpoint = null) {
+	log.info "refresh called: endpint '${endpoint}' "
+	if(endpoint) {
+		secureEncap(zwave.basicV1.basicGet(), endpoint)
+	} else {
+		def commands = []
+		commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1)
+		commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 5)
+		sendCommands(commands,1000)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd){
+	log.info "zwaveEvent ConfigurationReport called: ${cmd}"
+	def result = []
+	switch (cmd.parameterNumber) {
+		case 1:
+			state.tempOffset = cmd.scaledConfigurationValue
+			break
+		case 2:
+			state.humidityOffset = cmd.scaledConfigurationValue
+			break
+	}
+	result <<  createEvent(name: "configReport", value: "${state.tempOffset}, ${state.humidityOffset}%")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelEndPointReport cmd, endpoint = null) {
+	log.info "zwaveEvent MultiChannelEndPointReport called: cmd '${cmd}'"
+	if(!childDevices) {
+		def type = getNumberOfChildFromModel()
+		log.debug "MultiChannelEndPointReport: type '${type}'"
+		if (type) {
+			addChildSwitches(type)
+		} else {
+			log.debug "child endpoint=$cmd.endPoints"
+			addChildSwitches(cmd.endPoints)
+		}
+	}
+}
+
+def childOn(deviceNetworkId) {
+	log.info "childOn called: deviceNetworkId '${deviceNetworkId}'"
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand onOffCmd(0xFF, switchId)
+}
+
+def childOff(deviceNetworkId) {
+	log.info "childOff called: deviceNetworkId '${deviceNetworkId}', value '${value}'"
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) sendHubCommand onOffCmd(0x00, switchId)
+}
+
+
+private sendCommands(cmds, delay=1000) {
+	log.info "sendCommands called: cmds '${cmds}', delay '${delay}'"
+	sendHubCommand(cmds, delay)
+}
+
+private commands(commands, delay=200) {
+	log.info "commands called: commands '${commands}', delay '${delay}'"
+	delayBetween(commands.collect{ command(it) }, delay)
+}
+
+private secureEncap(cmd, endpoint = null) {
+	log.info "secureEncap called"
+
+	def CmdEncap = []
+	if (endpoint) {
+		CmdEncap = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:endpoint).encapsulate(cmd)
+	} else {
+		CmdEncap = cmd
+	}
+
+	if (zwaveInfo?.zw?.contains("s")) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(CmdEncap).format()
+	} else if (zwaveInfo?.cc?.contains("56")){
+		zwave.crc16EncapV1.crc16Encap().encapsulate(CmdEncap).format()
+	} else {
+		CmdEncap.format()
+	}
+	
+}
+
+private changeSwitch(endpoint, value) {
+	log.info "changeSwitch called: vlaue: '${value}', endpoint: '${endpoint}'"
+	def result = []
+	if(endpoint) {
+		String childDni = "${device.deviceNetworkId}/$endpoint"
```

Comment:
modern convention is to use `:` to separate endpoints in DNIs

Reply:
Thanks. I will update this line from "/" to ":".

String childDni = "${device.deviceNetworkId}/$endpoint"
==> String childDni = "${device.deviceNetworkId}:$endpoint"